feat: update secretTemplate documentation (#634)
cert-manager/website: content/en/docs/usage/certificate.md

spec:
  # Secret names are always required.
  secretName: example-com-tls

  # Secret template is optional. If set, these annotations
  # and labels will be copied to the secret named example-com-tls.

>   # Note: Once created, labels and annotations are not yet removed

Author: Added this note to keep clear we currently don't remove annotations from the secret if secretTemplate changes.

Reviewer: Is it also necessary to point out that if added or changed they are only updated when the cert is renewed?

Author: Thanks, @benlangfeld, updated as suggested.

>   # Note: Once created, labels and annotations are not yet removed
>   # from the Secret if removed from the secretTemplate.
>   # See https://github.com/jetstack/cert-manager/issues/4292.

Reviewer 2: Suggestion:
```suggestion
  # Note: Labels and annotations from the template are only synced to the Secret at the time when the certificate 
  is created or renewed. Currently labels and annotations can only be added, but not removed. Removing any 
  labels or annotations from the template or removing the template itself will have no effect.
  # See https://github.com/jetstack/cert-manager/issues/4292.
```

spec:
  # Secret names are always required.
  secretName: example-com-tls

  # Secret template is optional. If set, these annotations
  # and labels will be copied to the secret named example-com-tls.

  # Note: Labels and annotations from the template are only synced to the Secret at the time when the certificate 

>   is created or renewed. Currently labels and annotations can only be added, but not removed. Removing any 
>   labels or annotations from the template or removing the template itself will have no effect.

Reviewer: ```suggestion
  # is created or renewed. Currently labels and annotations can only be added, but not removed. Removing any 
  # labels or annotations from the template or removing the template itself will have no effect.
```

`example-com-tls` in the same namespace as the `Certificate` once the issuer has
successfully issued the requested certificate.

> If `secretTemplate` is present, annotations and labels set in this property
> will be copied over to `example-com-tls` secret. Both properties are optional.

Reviewer 3: Just wondering, how do we let people know that this is a feature that only exists since 1.5? The user would have to look for differences between 1.4 and 1.5 to find that out.

I'd go with smth like:

```suggestion
If `secretTemplate` _(since cert-manager 1.5)_ is present, annotations and labels set in this property
will be copied over to `example-com-tls` secret. Both properties are optional.
```

@irbekrm what do you think? 😅

Reviewer 4: Our docs are versioned so I don't think this is too much of a concern.

Reviewer 3: @JoshVanL It is very frustrating to have to select one by one the documentation versions to find when a feature was introduced and know if your version of cert-manager works with that feature.

I am often in the situation where I start using a feature and realize later on that the feature was introduced in a later version.

Reviewer 2: I guess we may want to merge this into release-next branch, not into master though?

Author: updated and rebased @irbekrm

Author: I've just added a release note based on the PR description and code comments, let me know if it needs to be rewritten. @benlangfeld I took the liberty to add your name to the contributors' list above.

Reviewer 5: Looks like our stupid spellchecker doesn't like the github usernames. You'll have to add them to the .spelling file. See https://github.com/cert-manager/website/blame/master/.spelling#L265

Author: Usernames added to .spelling.

Reviewer: @jonathansp I'm flattered, but I didn't do anything nearly worth being included in a list of contributors :)
